src/lib/api.js

```diff
@@ -133,8 +170,28 @@ const api = {
      * @return {Promise} HTTP response
      */
     put(url, data, options = {}) {
-        return api.xhr(url, { method: 'put', data, ...options });
+        return this.xhr(url, { method: 'put', data, ...options });
     }
-};
 
-export default api;
+    /**
+     * Wrapper function for XHR post put and delete
+     *
+     * @private
+     * @param {string} url - The URL for XHR
+     * @param {Object} options - The request options
+     * @return {Promise} - XHR promise
+     */
+    xhr(url, options = {}) {
+        let transformResponse;
+
+        if (options.responseType === 'text') {
+            transformResponse = transformTextResponse;
+        }
+
+        return this.client(url, filterOptions({ transformResponse, ...options }))
+            .then(parseResponse)
+            .catch(handleError);
+    }
+}
+
+export default new Api();
```

Comment:
I believe this will create a singleton shared by all instances of Preview. We should probably let Preview.js instantiate its api, instead.

Reply:
I think we want all calls to use the interceptors, but not pollute the global Axios instance. There are quite a few places that use the api outside of the main `Preview.js`, if we we stop sharing the instance, all calls will have their own Axios instance and need to have the interceptors passed to them. I would say this is not too far from what Preview previously had, without polluting the Axios instance.